omniscidb/QueryBuilder/QueryBuilder.cpp

```diff
@@ -828,7 +828,7 @@ BuilderExpr BuilderExpr::cast(const Type* new_type) const {
       return {builder_, expr_->cast(new_type), "", true};
     }
   } else if (expr_->type()->isDate()) {
-    if (new_type->isDate() || new_type->isTimestamp()) {
+    if (new_type->isNumber() || new_type->isDate() || new_type->isTimestamp()) {
```

Comment:
`isNumber` includes decimals and FP types. Is it intentional?

Reply:
I took conversion `timestamp` as an example, it uses 'timestamp' -> 'number, date, timestamp'. Maybe we should create some conversion map?